### ISSUE TITLE
Support x:Shared="false" for resources

### ISSUE THREAD

**Issue by kekekeks** [AvaloniaUI/Avalonia#2735]
https://docs.microsoft.com/en-us/dotnet/framework/xaml-services/x-shared-attribute

**Comment by Michail2732** [comment]
Hi, has any moves around this topic ?

**Comment by maxkatz6** [comment]
@Michail2732 why do you need this feature?
There is a support for templates in the resources.
https://github.com/AvaloniaUI/Avalonia/pull/6665
https://github.com/AvaloniaUI/Avalonia/issues/3817

**Comment by Michail2732** [comment]
@maxkatz6 hi i have some geometry paths in Xaml that i use in many places, without "x:Shared=false" i get exception like this "...already has a visual parent". This is expected since it is stored in a single instance in the ResourceDictionary. In WPF, this problem was solved by using "x:Shared=false".

**Comment by maxkatz6** [comment]
@Michail2732 what type is the geometry path? You can reuse StreamingGeometry instance. Or Geometry/GeometryGroup... But you can't store control there, unless it's a template.

**Comment by Michail2732** [comment]
Geometry is the "ViewBox->Canvas->Path" for example:
`<Viewbox x:Key="viewbox_icon_calendar" Stretch="Uniform">
		<Canvas Width="14.819" Height="14.819" Background="Transparent">
			<Canvas.RenderTransform>
				<TranslateTransform X="0" Y="0"/>
			</Canvas.RenderTransform>
			<Path Fill="{Binding $parent[2].Foreground}">
				<Path.Data>
					<PathGeometry Figures="M13.172 4.647a1.631 1.631 0 0 1 1.647 1.647v9.879a1.606 1.606 0 0 1-.477 1.153 1.563 1.563 0 0 1-1.169.494H1.647a1.58 1.58 0 0 1-1.153-.494A1.58 1.58 0 0 1 0 16.172V6.293A1.563 1.563 0 0 1 .494 5.124a1.606 1.606 0 0 1 1.153-.477H2.4V6.293H5.038V4.647H9.813V6.293h2.634V4.647Zm0 11.526V9.586H1.647v6.586ZM4.281 3V5.8H3.128V3ZM11.69 3V5.8H10.538V3Z" FillRule="NonZero"/>
				</Path.Data>
				<Path.RenderTransform>
					<TranslateTransform X="0" Y="-3"/>
				</Path.RenderTransform>
			</Path>
		</Canvas>
	</Viewbox>`

**Comment by maxkatz6** [comment]
PathGeometry can be put into the resources. You also can put TranslateTransform on it as well directly (PathGeometry.Transform property), so it also can be shared.

**Comment by Michail2732** [comment]
@maxkatz6 yes, I didn't think about that, I'm stupid. Thanks brat

**Comment by aldelaro5** [comment]
@maxkatz6 hey, I stumbled upon this issue because I thought I could make use of it even if I could do something similar to what the behaviors lib is doing, but I am very confused why this issue was closed because the merit here isn't necessarily to add the feature, but to simplify using it so that I can force my resources to be instantiated each time (which happens to be what I wanted).

I just question why this was closed because wouldn't the merit of implementing this be to simplify using this? Currently, I need to make my own type just to have a collection template which works, but is a bit verbose vs putting an attribute in the resource that would do this for me.
